cockatrice/src/tab_game.cpp

```diff
@@ -1814,10 +1814,16 @@ void TabGame::createMessageDock(bool bReplay)
         sayEdit->setCompleter(completer);
         actCompleterChanged();
 
-        if (spectator && !gameInfo.spectators_can_chat() && tabSupervisor->getAdminLocked()) {
+        /* Spectators can only talk if:
+         * (a) the game creator allows it
+         * (b) the spectator is a moderator/administrator
+         * (c) the spectator is a judge
+         */
+        if (!(spectator && (gameInfo.spectators_can_chat() || !tabSupervisor->getAdminLocked() || judge))) {
```

Comment:
I think the style this was in previously was a lot more readable, simply change this to && !judge?

Reply:
That won't work, because you need to be a spectator and one of the three options. This was the cleanest way I could come up with.

`if NOT (A & (B OR (NOT C) OR D))`

Reply:
I got confused so I made a truth table, your code will hide the chat for everyone who isn't a spectator :(
`(a, b, c, d, not (a and (b or not c or d)), a and not b and (c or not d)) for a, b, c, d in ((bool(i & (1 << n)) for n in range(4)) for i in range(1 << 4))`